chia/full_node/block_store.py

```diff
@@ -695,15 +685,17 @@ async def get_block_records_in_range(
         """
 
         ret: Dict[bytes32, BlockRecordDB] = {}
+        all_block_bytes: Dict[bytes32, memoryview] = {}
         if self.db_wrapper.db_version == 2:
             async with self.db_wrapper.reader_no_transaction() as conn:
                 async with conn.execute(
-                    "SELECT header_hash, block_record FROM full_blocks WHERE height >= ? AND height <= ?",
+                    "SELECT header_hash, block_record,block FROM full_blocks WHERE height >= ? AND height <= ?",
                     (start, stop),
                 ) as cursor:
                     for row in await cursor.fetchall():
                         header_hash = bytes32(row[0])
                         ret[header_hash] = BlockRecordDB.from_bytes(row[1])
+                        all_block_bytes[header_hash] = zstd.decompress(row[2])
```

Comment:
same comment here. I think you should throw away the full block bytes as early as possible by just picking out the plot filter info you need inside the loop